### PR TITLE
Fix PVC.Status.Capacity and AccessModes after binding

### LINKED ISSUE #29425: kubectl get pvc shows 0 capacity for PersistentVolumeClaim
Bound  PersistentVolumeClaim shows 0 capacity

```
$ gcloud compute disks list
NAME                                 ZONE            SIZE_GB  TYPE         STATUS
disk100-001                          europe-west1-d  100      pd-standard  READY
gke-unrest-micro-pool-199acc6c-3p31  europe-west1-d  100      pd-standard  READY
gke-unrest-micro-pool-199acc6c-4q55  europe-west1-d  100      pd-standard  READY

$ kubectl get pv
NAME             CAPACITY   ACCESSMODES   STATUS    CLAIM                           REASON    AGE
pv-disk100-001   100Gi      RWO           Bound     default/graphite-statsd-claim             2m

$ kubectl get pvc
NAME                    STATUS    VOLUME           CAPACITY   ACCESSMODES   AGE
graphite-statsd-claim   Bound     pv-disk100-001   0                        3m
```

From http://stackoverflow.com/questions/38463725/persistentvolumemound-bound-but-with-0-capacity-using-kubernetes-on-google-compu

### PR DESCRIPTION
**Pull request by jsafrane**
Also, fix unit tests to have the same claim and volume sizes in most of the
tests where we don't test matching based on size and test for a specific size
when we do actually test the matching.

Fixes #29425

### CODE REVIEW

File: pkg/controller/volume/persistentvolume/controller.go [hunk 1]
```diff
@@ -517,14 +518,11 @@ func (ctrl *PersistentVolumeController) syncVolume(volume *api.PersistentVolume)
...
 
-// updateClaimPhase saves new claim phase to API server.
-func (ctrl *PersistentVolumeController) updateClaimPhase(claim *api.PersistentVolumeClaim, phase api.PersistentVolumeClaimPhase) (*api.PersistentVolumeClaim, error) {
-	glog.V(4).Infof("updating PersistentVolumeClaim[%s]: set phase %s", claimToClaimKey(claim), phase)
-	if claim.Status.Phase == phase {
-		// Nothing to do.
-		glog.V(4).Infof("updating PersistentVolumeClaim[%s]: phase %s already set", claimToClaimKey(claim), phase)
-		return claim, nil
-	}
+// updateClaimStatusWithEvent saves new claim.Status to API server.
```
Comment (saad-ali) [inline pkg/controller/volume/persistentvolume/controller.go]: `updateClaimStatus` not `updateClaimStatusWithEvent`

File: pkg/controller/volume/persistentvolume/controller.go [hunk 2]
```diff
@@ -517,14 +518,11 @@ func (ctrl *PersistentVolumeController) syncVolume(volume *api.PersistentVolume)
...
+func (ctrl *PersistentVolumeController) updateClaimStatus(claim *api.PersistentVolumeClaim, phase api.PersistentVolumeClaimPhase, volume *api.PersistentVolume) (*api.PersistentVolumeClaim, error) {
```
Comment (saad-ali) [inline pkg/controller/volume/persistentvolume/controller.go]: As far as I can tell `updateClaimStatus(...)` is called with `volume *api.PersistentVolume` set to `nil` everywhere except by `updateClaimStatusWithEvent(...). And `updateClaimStatusWithEvent(...) is called with `volume *api.PersistentVolume` nil everywhere. So is volume ever not nil?

File: pkg/controller/volume/persistentvolume/controller.go (hunk 1 above)
Comment (saad-ali) [inline pkg/controller/volume/persistentvolume/controller.go]: Please add comment explaining what the parameters are and how they are used.

File: pkg/controller/volume/persistentvolume/controller.go (hunk 2 above)
Comment (jsafrane) [inline pkg/controller/volume/persistentvolume/controller.go]: `bind()` calls `updateClaimStatus` with non-nil volume.

I agree that `updateClaimStatusWithEvent` is always called with `nil`, however I'd prefer to keep the volume explicitly there for sake of consistency.

File: pkg/controller/volume/persistentvolume/controller.go (hunk 1 above)
Comment (jsafrane) [inline pkg/controller/volume/persistentvolume/controller.go]: comments added

File: pkg/controller/volume/persistentvolume/controller.go (hunk 2 above)
Comment (saad-ali) [inline pkg/controller/volume/persistentvolume/controller.go]: Ack

**Comment by saad-ali** [comment]
LGTM

**Comment by saad-ali** [comment]
Please squash commits and add LGTM label

**Comment by jsafrane** [comment]
squashed and pushed

**Comment by saad-ali** [comment]
@jsafrane Please cherry-pick this to the 1.3 release branch. 

CC @fabioy

**Pull request merged**